platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1092,4 +1112,18 @@ public void computeHash() {
             Thread.currentThread().interrupt();
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createSnapshot(@NonNull final Path targetPath) {
+        throwIfMutable();
+        throwIfDestroyed();
+        final long startTime = time.currentTimeMillis();
+        MerkleTreeSnapshotWriter.createSnapshot(this, targetPath);
+        if (snapshotMetrics != null && time != null) {
```

Comment:
IMO we should remove these null checks, and just update the metric. If the metric call fails because the object wasn't correctly initialized, then that's fine. 

Or, the alternative would be to do the null check, and throw an exception explaining what happened if one of these is null. What I don't think we should do is quitely continue operating, when something that definitely shouldn't be null *is* null

Reply:
okay, I removed the null checks from here. Also, I added a no-arg constructor for `MerkleRootSnapshotMetrics`. If the objects is created with it, it's no-op.